File: mobile/android/app/src/main/java/com/peakadex/app/core/api/ApiError.kt
package com.peakadex.app.core.api

sealed class ApiError(message: String) : Exception(message) {
    data object Unauthorized : ApiError("Session expired. Please log in again.")
    data object NotFound : ApiError("Resource not found.")
    data class ServerError(val statusCode: Int) : ApiError("Server error ($statusCode).")
    data class NetworkError(val rootCause: Throwable) : ApiError("Network error: ${rootCause.message}")
    data class DecodingError(val rootCause: Throwable) : ApiError("Failed to parse response: ${rootCause.message}")
    data object Unknown : ApiError("An unexpected error occurred.")
}

File: mobile/ios/Peakadex/Core/API/APIError.swift
import Foundation

enum APIError: LocalizedError {
    case unauthorized
    case notFound
    case serverError(statusCode: Int)
    case decodingFailed(Error)
    case networkFailed(Error)
    case unknown

    var errorDescription: String? {
        switch self {
        case .unauthorized:               return "Session expired. Please log in again."
        case .notFound:                   return "Resource not found."
        case .serverError(let code):      return "Server error (\(code))."
        case .decodingFailed(let error):  return "Failed to parse response: \(error.localizedDescription)"
        case .networkFailed(let error):   return "Network error: \(error.localizedDescription)"
        case .unknown:                    return "An unexpected error occurred."
        }
    }
}
